packages/web/src/services/track-download.ts

```diff
@@ -10,29 +14,57 @@ function isMobileSafari() {
   )
 }
 
+function browserDownload({
+  url,
+  filename
+}: {
+  url: string
+  filename?: string
+}) {
+  if (document) {
+    const link = document.createElement('a')
+    link.href = url
+    // taget=_blank does not work on ios safari and will cause the download to be
+    // unresponsive.
+    if (!isMobileSafari()) {
+      link.target = '_blank'
+    }
+    link.download = filename ?? ''
+    link.click()
+    link.remove()
+  } else {
+    throw new Error('No document found')
+  }
+}
+
 class TrackDownload extends TrackDownloadBase {
-  async downloadTrack({ url, filename }: { url: string; filename: string }) {
-    const response = await window.fetch(url)
-    if (!response.ok) {
+  async downloadTrack({ files, rootDirectoryName }: DownloadTrackArgs) {
+    const responsePromises = files.map(
+      async ({ url }) => await window.fetch(url)
+    )
+    const responses = await Promise.all(responsePromises)
```

Comment:
Just confirming: `Promise.all` will short circuit to rejection as soon as any of the promises in the array rejects, causing the whole operation to fail. That seems like what we want?